### DISCUSSION TITLE
AMD Vulkan | Fixed

### DISCUSSION THREAD

**Discussion by ghost** [YimMenu/HorseMenu#195] [General]
I saw a while ago HorseMenu had issues with AMD GPU's and bought one specifically for this reason. I've managed to find a workaround although currently not optimized very well, it works!

https://youtu.be/lDxrWvAHxyU?si=eVaDBxRaKTeCJsMr

Anyways, the main thingy with this is, it doesn't actually hook anything. It's an overlay which has its pros and cons. Now my question to you is, should I put time and effort into perfecting this to contribute this to HorseMenu or do the cons outweigh the pros?

I'm gonna list them out for you:

Pros:
- DX11 Overlay so rendering textures is much much easier
- Can be reinjected more than 2 times unlike DX12 and Vulkan API
- Can be invisible to OBS if the user wants it to be (good for sneaky streamers)
- Way better performance
- Would replace both the DX12 and Vulkan renderer, reducing the file to like 300 lines

Cons: 
- It is an overlay, this means you cannot run the game in full screen, only in windowed or borderless
- Tabbing out of the game makes it so the menu is invisible (intentional)
- It only works if it is the foreground window, I have not tested this yet but it could be that the discord overlay and this one would fight for which one would be on top. 
- ImGui inputs have to be replicated

Now as you can see in the video the mouse is glitchy, I already have a fix for this, but it will take me some time to perfect it hence this post to see if it is in fact worth it.

**Comment by Rxann** [comment]
This is some very good research. Thank you for the contribution and I will look into it later.
